src/Hyperion/SerializerOptions.cs

```diff
@@ -72,7 +80,25 @@ public class SerializerOptions
         internal readonly List<Func<string, string>> CrossFrameworkPackageNameOverrides =
             DefaultPackageNameOverrides();
 
-        public SerializerOptions(bool versionTolerance = false, bool preserveObjectReferences = false, IEnumerable<Surrogate> surrogates = null, IEnumerable<ValueSerializerFactory> serializerFactories = null, IEnumerable<Type> knownTypes = null, bool ignoreISerializable = false, IEnumerable<Func<string, string>> packageNameOverrides = null)
+        [Obsolete]
+        public SerializerOptions(
```

Comment:
This is the old constructor, added back in for backward compatibility issue.
New constructor broke backward compatibility because it was called using reflection and `Activator` class.

Reply:
IMHO, this is part of the problem with using optional parameters everywhere - very difficult to properly version the constructor when that happens. That's a design we inherited, but we might want to rethink that.